sklearn/metrics/_regression.py

```diff
@@ -99,15 +107,22 @@ def _check_reg_targets(y_true, y_pred, multioutput, dtype="numeric"):
         just the corresponding argument if ``multioutput`` is a
         correct keyword.
     """
+    if xp is None:
+        input_arrays = [y_true, y_pred]
+        if multioutput is not None and not isinstance(multioutput, str):
+            input_arrays.append(multioutput)
+
+        xp, _ = get_namespace(*input_arrays)
```

Comment:
This part is basically copy pasted everywhere. Can this not be reduced into:

```suggestion
    xp, _ = get_namespace(y_true, y_pred, multioutput)
```

and let `get_namespace` ignore everything which is `None` or a `str`?